src/components/BusinessDonationHistory/BusinessDonationHistory.jsx

```diff
@@ -11,17 +25,45 @@ const BusinessDonationHistory = () => {
   const [selectedDonationId, setSelectedDonationId] = useState(null);
   const TABLE_HEADERS = ['Submitted By', 'Food Bank', 'Date', 'Action'];
 
+  // for pagination
+  const [currentTotalDonationNum, setCurrentTotalDonationNum] = useState(0);
+  const [currentPageNum, setCurrentPageNum] = useState(1);
+  const [pageLimit, setPageLimit] = useState(1);
+
+  // for search
+  const [searchInput, setSearchInput] = useState('');
+
+  // TEMPORARY DISABLE LINE SO THAT IN THE FUTURE IF YOU WANT TO ADD
+  // A DROP DOWN TO CHANGE PAGINATION NUMBER JUST USE THE SET FUNCTION
+  // AND REMOVE THE COMMENT
+  // eslint-disable-next-line no-unused-vars
+  const [paginationNumber, setPaginationNumber] = useState(13);
+
   useEffect(() => {
     const getData = async () => {
       try {
-        const response = await backend.get('/donation');
+        const response =
+          searchInput === ''
+            ? await backend.get(
+                `/donation/?donationsLimit=${paginationNumber}&pageNum=${currentPageNum}`,
+              )
+            : await backend.get(
+                `/donation/search/${searchInput}?donationsLimit=${paginationNumber}&pageNum=${currentPageNum}`,
+              );
+
         setData(response.data);
+
+        const donationNumResponse = await backend.get(
+          `/donation/totalDonations/${searchInput === '' ? '' : searchInput}`,
```

Comment:
`/donation/totalDonations/${searchInput}`
This works the same way as what you have currently